### DISCUSSION TITLE
V3 Local Secret Key must be 32 bytes long symmetric key Error

### DISCUSSION THREAD

**Discussion by Jacobinoo** [panva/paseto#32] [Q&A] [answered]
Hi! My secret key was generated using PHP Paseto with openssl_random_pseudo_bytes(32) and then converted from bin to hex.
I don't understand why my secret key returns an error that the length is invalid. It was generated using 32 bytes.

```js
const paseto = require('paseto');
const { V3: { decrypt } } = paseto;

(async () => {
  {
    const payload = await decrypt("example_token", "b244ac595fbe3a6ea8c3fad93f66d15221121428fd03dcccf32203e364f504ed")
  }
})()
```

**Comment by Jacobinoo** [comment]
By the way: is the required length editable somehow??

**Reply by panva** [reply]
No. This is dictated by the PASETO specification.

**Comment by panva** [comment, marked as answer]
"b244ac595fbe3a6ea8c3fad93f66d15221121428fd03dcccf32203e364f504ed" is 64 octets, not 32, you need to decode your hex encoded string into a 32 byte buffer first.

```
Buffer.from(hexString, 'hex')
```

**Reply by Jacobinoo** [reply]
Sorry, how exactly should this look like? I don't know what code should I write for this to work.

**Reply by Jacobinoo** [reply]
Well, I figured it out somehow!
Solution:
```js
const paseto = require('paseto');
const { V3: { decrypt } } = paseto;
const secretKey = createSecretKey(new Buffer.from("secret_hexadecimal_key", "hex"));
(async () => {
  {
    const payload = await decrypt("example_token", secretKey)
  }
})()
```
